### PR TITLE
Add new exercise: Binary search

### PR DESCRIPTION
**Pull request by rchavarria**
This PR adds a new exercise from the xjavascript track.

### CODE REVIEW

File: binary-search/example.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if (array[i] < array[i - 1]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+function recursiveSearch(array, value, start, end) {
+  if (start == end) return -1;
```
Comment (matthewmorgan) [inline binary-search/example.js]: Should this use `===` instead of `==`?

**Comment by matthewmorgan** [comment]
@rchavarria I am noticing the test order actual/expected is reversed on most of the tests.

File: binary-search/example.js (hunk 1 above)
Comment (rchavarria) [inline binary-search/example.js]: Totally agree

**Comment by rchavarria** [comment]
You're right. I didn't notice that.

**Pull request merged**